.changes/next-release/bugfix-configprovider-46546.json

```diff
@@ -0,0 +1,5 @@
+{
+  "type": "bugfix",
+  "category": "configprovider",
+  "description": "Fix bug when deep copying config value store where overrides were not preserved"
```

Comment:
For this bug fix one, yes it technically is. Specifically the `config_store` is a public interface via the `Session.get_component()` API. So a user could deepcopy the original `config_store` on the session, update it, and reattach it to the session. The bug is that the deep copy would not have the same default overrides that were provided to the original config store set here: https://github.com/boto/botocore/blob/a587e9261d11473e10491a1b09a55a0148951334/botocore/session.py#L255




For the `enhancement` one, we don't believe it will impact end users. When creating a client, the `config_store` is only used at instantiation of the client so providing a copy of it instead to the client should not impact an end to end user (i.e. it is not possible to update the `config_store` on the session and expected the change to permeate to a preexisting client).

The purpose of adding a changelog entry for it was more for awareness/acknowledge that we did make this subtle change if you are more knowledgeable on the architecture of botocore. It was labeled as an enhancement because there are not really any other categories it fits into. The argument for calling it an enhancement was that the client's config store is now more inline with the client's event emitters where they are snapshot copies of the original config_store/event_emitter from the session. I can go either way on a changelog for this or not.